### PR TITLE
Provider destroy not to destroy dependent manager automatically

### PR DESCRIPTION
**Pull request by jameswnl**
Depending on https://github.com/ManageIQ/manageiq/pull/16755

### CODE REVIEW

**Comment by jameswnl** [comment]
@miq-bot add_lables gaprindashvili/yes, bug, blocker

**Comment by jameswnl** [comment]
@miq-bot add_labels gaprindashvili/yes, bug, blocker

File: spec/support/ansible_shared/provider.rb [hunk 1]
```diff
@@ -22,10 +22,12 @@
   describe "#destroy" do
     it "will remove all child objects" do
       subject.automation_manager.configured_systems = [
-        FactoryGirl.create(:configured_system, :computer_system =>
-          FactoryGirl.create(:computer_system,
+        FactoryGirl.create(:configured_system_automation_manager,
```
Comment (bdunne) [inline spec/support/ansible_shared/provider.rb]: What is this factory?  We don't have a ConfiguredSystemAutomationManager model or table.

Comment (bdunne) [reply spec/support/ansible_shared/provider.rb]: I also don't see it in the factories in manageiq

Comment (jameswnl) [reply spec/support/ansible_shared/provider.rb]: This PR depends on https://github.com/ManageIQ/manageiq/pull/16755 which (other than the `destroy` method rewrite) has created the factory `:configured_system_automation_manager` in [here](https://github.com/ManageIQ/manageiq/pull/16755/files#diff-3e18369d4c9d44e9491f3507daaa2b08R6)

**Comment by blomquisg** [comment]
@jameswnl I'm guessing this is failing specs because of the missing dependent changes in the core MIQ repo?

**Comment by jameswnl** [comment]
@blomquisg yes, it depends on that core miq PR.
And yes, `wip` because of that.

**Comment by jameswnl** [comment]
@miq-bot remove_label wip

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 0575317aa27d008097acd86ac47b762f1d181b75
Author: Adam Grare <agrare@redhat.com>
Date:   Wed Jan 17 16:18:35 2018 -0500

    Merge pull request #49 from jameswnl/destroy-provider
    
    Provider destroy not to destroy dependent manager automatically
    (cherry picked from commit 29b7146fa830a2d47333ec9cde751ffb6073726d)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1536039
    https://bugzilla.redhat.com/show_bug.cgi?id=1536041
```

**Pull request merged**